plugins/file/plugin.go

```diff
@@ -125,24 +206,120 @@ func setupFile6(args ...string) (handler.Handler6, error) {
 }
 
 func setupFile4(args ...string) (handler.Handler4, error) {
-	log.Print("plugins/file: loading `file` plugin for DHCPv4")
-	return nil, nil
+	_, h4, err := setupFile(false, args...)
+	return h4, err
 }
 
 func setupFile(v6 bool, args ...string) (handler.Handler6, handler.Handler4, error) {
-	if len(args) < 1 {
-		return nil, nil, errors.New("plugins/file: need a file name")
+	if v6 {
+		if len(args) < 1 {
+			return nil, nil, errors.New("plugins/file: need a file name")
+		}
+		filename := args[0]
+		if filename == "" {
+			return nil, nil, errors.New("plugins/file: got empty file name")
+		}
+		records, err := LoadDHCPv6Records(filename)
+		if err != nil {
+			return nil, nil, fmt.Errorf("plugins/file: failed to load DHCPv6 records: %v", err)
+		}
+		log.Printf("plugins/file: loaded %d leases from %s", len(records), filename)
+		StaticRecords = records
+	} else {
+		if len(args) < 3 {
+			return nil, nil, errors.New("plugins/file: need a file name, server IP, netmask and a lease time")
+		}
+		filename = args[0]
+		if filename == "" {
+			return nil, nil, errors.New("plugins/file: got empty file name")
+		}
+		var err error
+		_, network, err = net.ParseCIDR(args[1])
+		if err != nil {
+			return Handler6, Handler4, errors.New("plugins/file: expected an IPv4 address, got: " + args[1])
+		}
+		LeaseTime, err = time.ParseDuration(args[2])
+		if err != nil {
+			return Handler6, Handler4, errors.New("plugins/file: expected an uint32, got: " + args[2])
+		}
+		records, err := LoadDHCPv4Records(filename)
+		if err != nil {
+			return nil, nil, fmt.Errorf("plugins/file: failed to load DHCPv4 records: %v", err)
+		}
+		DHCPv4Records = records
+
+		rand.Seed(time.Now().Unix())
+
+		log.Printf("plugins/file: loaded %d leases from %s", len(DHCPv4Records), filename)
+	}
+
+	return Handler6, Handler4, nil
+}
+func createIP(network *net.IPNet) (Record, error) {
```

Comment:
I think ti is better if this returns a reference to a Record and avoid copying the whole struct.